### PR TITLE
Fix #1472: Investigate @Scheduled methods with @Transactional annotation

### LINKED ISSUE #1472: Investigate @Scheduled methods with @Transactional annotation
We should review the behavior of the combination of scheduling and transactional behavior, just to be sure there could be no strange side effects.

### PR DESCRIPTION
**Pull request by banterCZ**
In general, the usage of `@Scheduled` and `@Transactional` together over a method should work. But let's improve the design, split them into separate layers, and isolate responsibilities.

### CODE REVIEW

File: powerauth-java-server/src/main/resources/application-dev.properties [hunk 1]
```diff
@@ -3,3 +3,6 @@ spring.liquibase.enabled=true
 spring.liquibase.change-log=classpath:db/changelog/db.changelog-module.xml
 
 powerauth.service.scheduled.job.fido2AuthenticatorCacheEviction=60000
+
+logging.level.com.wultra=DEBUG
+logging.level.io.getlime=DEBUG
```
Comment (romanstrobl) [inline powerauth-java-server/src/main/resources/application-dev.properties:8]: Is this change intentional?

**Review by romanstrobl — APPROVED** [review]
Looks OK.

**Review by jnpsk — APPROVED** [review]
nice!
